cmd.c

```diff
@@ -0,0 +1,1842 @@
+
+#include "buffer.h"
+#include "cmd.h"
+#include "logging.h"
+#include "types.h"
+#include <assert.h>
+
+//Function declarations
+CMD_FUN (quit);
+CMD_FUN (close);
+CMD_FUN (log);
+CMD_FUN (set);
+CMD_FUN (server);
+CMD_FUN (reconnect);
+CMD_FUN (disconnect);
+CMD_FUN (realsname);
+CMD_FUN (join);
+CMD_FUN (rejoin);
+CMD_FUN (part);
+CMD_FUN (unaway);
+CMD_FUN (away);
+CMD_FUN (afk);
+CMD_FUN (nick);
+CMD_FUN (topic);
+CMD_FUN (msg);
+CMD_FUN (ping);
+CMD_FUN (amsg);
+CMD_FUN (me);
+CMD_FUN (tab);
+CMD_FUN (sort);
+CMD_FUN (left);
+CMD_FUN (right);
+CMD_FUN (ignore);
+CMD_FUN (mode);
+CMD_FUN (cmd);
+CMD_FUN (help);
+
+//Number of Commands
+#define NCMDS 33
+
+//cmd_funcs can't be malloc'd the regular way because of -Werror
+//malloc returns a void pointer and with -Werror you can't
+//cast from a regular pointer to a function pointer. -Russell
```

Comment:
Could you not use `int (**cmd_funcs)(args...) = malloc(n * sizeof(int (*)(args...)))`?  A function pointer is data, so a pointer to function pointer should be a data pointer and thus castable from `void *`.  (Just tested a tiny program and it compiles happily.)
Agreed you can't malloc an array of functions, but an array of function pointers is fine.

"Any problem in Computer Science can be solved by another layer of indirection."